docs/source/tutorials/improving_ml_performance.ipynb

```diff
@@ -0,0 +1,3815 @@
+       "463                                          [384]   \n",
+       "540                                            [2]   \n",
+       "592                                          [260]   \n",
+       "706  [714, 707, 709, 710, 711, 712, 713, 708, 715]   \n",
+       "707  [714, 706, 709, 710, 711, 712, 713, 708, 715]   \n",
+       "708  [714, 707, 706, 709, 710, 711, 712, 713, 715]   \n",
+       "709  [714, 707, 706, 710, 711, 712, 713, 708, 715]   \n",
+       "710  [714, 707, 706, 709, 711, 712, 713, 708, 715]   \n",
+       "711  [714, 707, 706, 709, 710, 712, 713, 708, 715]   \n",
+       "712  [714, 707, 706, 709, 710, 711, 713, 708, 715]   \n",
+       "713  [714, 707, 706, 709, 710, 711, 712, 708, 715]   \n",
+       "714  [707, 706, 709, 710, 711, 712, 713, 708, 715]   \n",
+       "715  [714, 707, 706, 709, 710, 711, 712, 713, 708]   \n",
+       "\n",
+       "     distance_to_nearest_neighbor  \n",
+       "2                             0.0  \n",
+       "260                           0.0  \n",
+       "384                           0.0  \n",
+       "463                           0.0  \n",
+       "540                           0.0  \n",
+       "592                           0.0  \n",
+       "706                           0.0  \n",
+       "707                           0.0  \n",
+       "708                           0.0  \n",
+       "709                           0.0  \n",
+       "710                           0.0  \n",
+       "711                           0.0  \n",
+       "712                           0.0  \n",
+       "713                           0.0  \n",
+       "714                           0.0  \n",
+       "715                           0.0  "
+      ]
+     },
+     "execution_count": 15,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "exact_duplicates = full_duplicate_results[(full_duplicate_results[\"is_near_duplicate_issue\"] == True) & (full_duplicate_results[\"near_duplicate_score\"] == 0.0)].sort_values(\"near_duplicate_score\")\n",
+    "exact_duplicates"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9672b692-fc3e-4467-b5c4-4c367b55eb30",
+   "metadata": {},
+   "source": [
+    "To remove the exact duplicates that occur between our training and test sets from our training data, let's define the last index of data points in our training set. Then we'll drop rows from our training data that correspond to all indices less than or equal to our cutoff index that are also found in the set of exact duplicates flagged by `Datalab`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "id": "df1fa776-a89d-454f-a42d-cfd8b73f2f1b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define training index cutoff and find the exact duplicate indices to reference\n",
+    "train_idx_cutoff = len(preprocessed_train_data) - 1\n",
+    "exact_duplicates_indices = exact_duplicates.index\n",
```

Comment:
I think this needs to be subsetted to the set of exact_duplicates where at least one of the datapoints is from the test set.